File: examples/users_list.js
var Bot    = require('../index');
var AUTH   = 'auth+live+xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx';
var USERID = 'xxxxxxxxxxxxxxxxxxxxxxxx';
var ROOMID = 'xxxxxxxxxxxxxxxxxxxxxxxx';

var bot = new Bot(AUTH, USERID, ROOMID);

var theUsersList = { };

bot.on('roomChanged', function (data) {
  // Reset the users list
  theUsersList = { };

  var users = data.users;
  for (var i=0; i<users.length; i++) {
    var user = users[i];
    theUsersList[user.userid] = user;
  }
});

bot.on('registered', function (data) {
  var user = data.user[0];
  theUsersList[user.userid] = user;
});

bot.on('deregistered', function (data) {
  var user = data.user[0];
  delete theUsersList[user.userid];
});

File: examples/users_list.py
from ttapi import Bot

AUTH   = 'auth+live+xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
USERID = 'xxxxxxxxxxxxxxxxxxxxxxxx'
ROOMID = 'xxxxxxxxxxxxxxxxxxxxxxxx'
bot = Bot(AUTH, USERID, ROOMID)

theUsersList = { }

def roomChanged(data):
   global theUsersList
   # Reset the users list
   theUsersList = {}
   users = data['users']
   for user in users:
      theUsersList[user['userid']] = user


def registered(data):
   global theUsersList
   user = data['user'][0]
   theUsersList[user['userid']] = user


def deregistered(data):
   global theUsersList
   user = data['user'][0]
   del theUsersList[user['userid']]


bot.on('roomChanged',  roomChanged)
bot.on('registered',   registered)
bot.on('deregistered', deregistered)

bot.start()
